x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/rules_stats_collector.ts

```diff
@@ -0,0 +1,261 @@
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        benchmark_version: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.version',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        benchmark_name: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.benchmark.name',
+            },
+            size: 1,
+            sort: {
+              '@timestamp': 'desc',
+            },
+          },
+        },
+        passed_findings_count: {
+          filter: {
+            bool: {
+              filter: [
+                {
+                  bool: {
+                    should: [
+                      {
+                        term: {
+                          'result.evaluation': 'passed',
+                        },
+                      },
+                    ],
+                    minimum_should_match: 1,
+                  },
+                },
+              ],
+            },
+          },
+        },
+        failed_findings_count: {
+          filter: {
+            bool: {
+              filter: [
+                {
+                  bool: {
+                    should: [
+                      {
+                        term: {
+                          'result.evaluation': 'failed',
+                        },
+                      },
+                    ],
+                    minimum_should_match: 1,
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+    },
+  },
+
+  size: 0,
+  _source: false,
+});
+
+const getCspmRulesStats = (aggregatedRulesStats: RulesStats, logger: Logger): CspmRulesStats[] => {
+  const rules = aggregatedRulesStats.rules.buckets;
+
+  const cspmRulesStats = rules.map((rule) => ({
+    rule_id: rule.key,
+    rule_name: rule.rule_name.top[0].metrics['rule.name'],
+    rule_section: rule.rule_section.top[0].metrics['rule.section'],
+    rule_version: rule.rule_version.top[0].metrics['rule.version'],
+    rule_number: rule.rule_number.top[0].metrics['rule.benchmark.rule_number'],
+    posture_type: rule.posture_type.top[0].metrics['rule.benchmark.posture_type'],
+    benchmark_name: rule.benchmark_name.top[0].metrics['rule.benchmark.name'],
+    benchmark_id: rule.benchmark_id.top[0].metrics['rule.benchmark.id'],
+    passed_findings_count: rule.passed_findings_count.doc_count,
+    benchmark_version: rule.benchmark_version.top[0].metrics['rule.benchmark.version'],
+    failed_findings_count: rule.failed_findings_count.doc_count,
+  }));
+
+  return cspmRulesStats;
+};
+
+export const getRulesStats = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<CspmRulesStats[]> => {
+  try {
+    const isIndexExists = await esClient.indices.exists({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+    });
```

Comment:
seems like we check for this index in all our collectors. maybe we could check it once and use the result in all of them